### PR TITLE
[Merged by Bors] - Down with the system!

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Remove all the `.system()` possible.
- Check for remaining missing cases.

## Solution

- Remove all `.system()`, fix compile errors
- 32 calls to `.system()` remains, mostly internals, the few others should be removed after #2446

### CODE REVIEW

**Comment by cart** [comment]
It's so ... beautiful.

**Comment by alice-i-cecile** [comment]
What sort of internal `.system` calls do we have left that won't be cleaned up by #2446?

**Comment by Ratysz** [comment]
Hahah, I'm sitting on a very similar changeset. Couple differences: started on updating the docs and removed `IntoSystem` from the prelude, which soft-deprecates it. Wanted to also rewrite `make_parallel!()` into less of an atrocity.

> What sort of internal .system calls do we have left that won't be cleaned up by #2446?

Code that tests if a function is a system.

File: crates/bevy_app/src/app_builder.rs [hunk 1]
```diff
@@ -157,7 +157,7 @@ impl AppBuilder {
     /// Adds a system that runs every time `app.update()` is called by the runner
     ///
     /// Systems are the main building block in the Bevy ECS app model. You can define
-    /// normal rust functions, and call `.system()` to make them be Bevy systems.
+    /// normal rust functions, and call `` to make them be Bevy systems.
```
Comment (MinerSebas) [inline crates/bevy_app/src/app_builder.rs:160]: This needs to be rewritten

File: crates/bevy_ecs/src/query/fetch.rs [hunk 2]
```diff
@@ -708,7 +708,7 @@ impl<'w, T: Fetch<'w>> Fetch<'w> for OptionFetch<T> {
 ///         }
 ///     }
 /// }
-/// # print_moving_objects_system.system();
+/// # print_moving_objects_system;
```
Comment (MinerSebas) [inline crates/bevy_ecs/src/query/fetch.rs:711]: This usage in doc-tests shouldn't be removed, as it checks if this function is still a valid system.
This check is also done in many other doc-tests.

Comment (mockersf) [reply crates/bevy_ecs/src/query/fetch.rs]: oh forgot that pattern in bevy_ecs... thanks, I added them back

**Comment by mockersf** [comment]
> Wanted to also rewrite make_parallel!() into less of an atrocity.

Ended up doing that to remove the macro, but there's still an issue: system dependency cycles now stack overflow instead of panicking

**Comment by Ratysz** [comment]
I have no idea why that's happening... Have to sleep first, I guess.

**Comment by mockersf** [comment]
there was an infinite loop of two traits calling each other `name` method that needed a `.system()` to return something concrete at some point, fixed 👍 

53 `.system()` now!

**Comment by Ratysz** [comment]
Ah, yeah: that's not the `.system()` that needs to be removed, it's just named the same way. Weird it didn't come up sooner.

**Comment by Nilirad** [comment]
Well, don't mind me...

*proceeds to sneak in `.system()` in every doctest*

**Comment by cart** [comment]
This re-adds AppBuilder (now that we've removed it on main). Can you remove it?

**Comment by mockersf** [comment]
oh messed up my rebase and re-added it, it should be good now 👍

**Pull request closed without merging**